src/charts/heatmap/heatmap.directive.js

```diff
@@ -271,8 +286,20 @@ angular.module('patternfly.charts').directive('pfHeatmap', function ($compile, $
 
         //Adding events
         blocks.on('mouseover', function () {
+          var fillColor;
           blocks.call(highlightBlock, false);
-          d3.select(this).call(highlightBlock, true);
+          if (scope.maxSize < 16) {
+            // Get fill color for current block
+            fillColor = color(d3.select(this).map(function (d) {
+              return d[0].__data__.value;
+            }));
+            // Highlight all blocks matching fill color
+            blocks[0].forEach(function (block) {
+              highlightBlockColor(d3.select(block), fillColor);
+            });
+          } else {
```

Comment:
I believe we want to set the tooltip based on the range as well so that there is one tooltip for the entire range not per node. This would mean we need to ignore the tooltip field on the nodes and provide a means for the application to give us the tooltip based on the current range being hovered on.